python/MooseDocs/extensions/__init__.py

```diff
@@ -51,7 +51,10 @@ def create_collection(name, syntax, action, groups=[], **kwargs):
     children = []
     for group in groups:
         if action == 'object':
-            items = get_collection_items(syntax[group].objects(name, include_self=False), **kwargs)
+            # Alphabetize items before creating xml elements from them
+            items_to_collect = syntax[group].objects(name, include_self=False)
+            items_to_collect.sort(key=lambda item: item.name)
+            items = get_collection_items(items_to_collect, **kwargs)
```

Comment:
I kept it out of there and specifically in the "if action == 'object'" section just in case other things calling that method need them in a different order.  But If you don't think that's a problem at all, then I'll shift it in.